internal/pkg/pool/pool_test.go

```diff
@@ -0,0 +1,167 @@
+package pool
+
+import (
+	"bytes"
+	"compress/gzip"
+	"io"
+	"testing"
+
+	"github.com/c2h5oh/datasize"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBufferWriters(t *testing.T) {
+	t.Parallel()
+
+	rawData := []byte("foo")
+
+	pool := BufferedWriters(100 * datasize.KB)
+
+	var out1 bytes.Buffer
+	w1 := pool.WriterTo(&out1)
+	_, err := w1.Write(rawData)
+	require.NoError(t, err)
+	require.NoError(t, w1.Flush())
+	require.Equal(t, rawData, out1.Bytes())
+
+	var out2 bytes.Buffer
+	w2 := pool.WriterTo(&out2)
+	require.NotSame(t, w1, w2)
+	_, err = w2.Write(rawData)
+	require.NoError(t, err)
+	require.NoError(t, w2.Flush())
+	require.Equal(t, rawData, out2.Bytes())
+
+	// Put the writers back to the pool
+	pool.Put(w1)
+	pool.Put(w2)
+
+	// Writer is reused (w1), but it cannot be asserted
```

Comment:
`sync.Pool` vrati nahodne niektory volny objekt, preto sa to neda nejak overit.
Teda testujeme, ze to funguje, ale nevieme overit, ze sa dany objekt pouzil znova.